src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -960,17 +960,63 @@ public virtual TokenValidationResult ValidateToken(string token, TokenValidation
+                    return ValidateToken(token, null, null, validationParameters);
+                }
+            }
+            catch (Exception ex)
+            {
+                return new TokenValidationResult
+                {
+                    Exception = ex,
+                    IsValid = false
+                };
+            }
+
+        }
+
+        /// <summary>
+        ///  Private method for token validation, responsible for (1) obtaining a configuration from the <see cref="TokenValidationParameters.ConfigurationManager"/>,
+        ///  (2) revalidating using the Last Known Good Configuration (if present), and obtaining a refreshed configuration (if necessary) and revalidating using
+        ///  it.
+        /// </summary>
+        /// <param name="token">The JWS string, only to be supplied if the token is a JWS.</param>
+        /// <param name="outerToken">The outer token of the JWE, only to be supplied if the token is a JWE.</param>
+        /// <param name="decryptedJwt">The inner decrypted token of the JWE, only to be supplied if the token is a JWE.</param>
+        /// <param name="validationParameters">The <see cref="TokenValidationParameters"/> to be used for validation.</param>
+        /// <returns></returns>
+        private TokenValidationResult ValidateToken(string token, JsonWebToken outerToken, string decryptedJwt, TokenValidationParameters validationParameters)
+        {
+
             var validationParametersCopy = validationParameters;
-            if (validationParameters.ConfigurationManager != null)
+            if (validationParametersCopy.ConfigurationManager != null)
             {
                 try
                 {
+                    // TODO: Use overloads to get rid of the Clone() operation.
                     var configuration = validationParametersCopy.ConfigurationManager.GetBaseConfigurationAsync(CancellationToken.None).ConfigureAwait(false).GetAwaiter().GetResult();
                     validationParametersCopy = validationParameters.Clone();
                     validationParametersCopy.Configuration = configuration;
                 }
                 catch (Exception ex)
                 {
+                    // TODO: Get rid of this check and continue with validation regardless of the properties set on the TVP.
+                    // It's too hard to capture everything in this check, we should just fail later on in the validation process.
+
                     // If either a signing key or a valid issuer was not provided through the TVP, we want to stop processing on configuration retrieval failure.
                     // Otherwise, we should keep going with validation.
                     if ((validationParametersCopy.ValidateIssuer && string.IsNullOrWhiteSpace(validationParametersCopy.ValidIssuer) && validationParametersCopy.ValidIssuers.IsNullOrEmpty() && validationParametersCopy.IssuerValidator == null
```

Comment:
Yeah, I think the logic in this check isn't quite right. I was going to leave this for a future PR, but I've gone ahead and removed the check since I do think it's too difficult to capture all the relevant scenarios.